### PR TITLE
Remove separate treatment queries

### PR DESCRIPTION
**Pull request by nleanba**
See https://github.com/factsmission/synospecies/issues/128

Note that this is a significant and breaking API change.
**Do not merge without also updating synospecies.**

These Types have changed:
``` ts
export type TreatmentDetails = {
  materialCitations: MaterialCitation[];
  date?: number;
  creators?: string;
  title?: string;
};

export type Treatment = {
  url: string;
  details: Promise<TreatmentDetails>;
};

export type TaxonName = {
  uri: string;
  treatments: {
    aug: Set<Treatment>;
    cite: Set<Treatment>;
  };
  loading: boolean;
};

type Treatments = {
  def: Set<Treatment>;
  aug: Set<Treatment>;
  dpr: Set<Treatment>;
  cite: Set<Treatment>;
};
export type JustifiedSynonym = {
  taxonConceptUri: string;
  taxonName: TaxonName;
  taxonConceptAuthority?: string;
  justifications: JustificationSet;
  treatments: Treatments;
  loading: boolean;
};
```

The type `TreatmentSet` has been removed.

Also `SynonymGroup` has gained two new fields:
```ts
export default class SynonymGroup implements AsyncIterable<JustifiedSynonym> {
  ...
  treatments: Map<string, Treatment> = new Map();
  taxonNames: Map<string, TaxonName> = new Map();
  ...
}
```

They are primarily for internal use, and notably don't have their own async iterators, but they might still be useful for applications.
Both are only updated when a new taxon concept is found.

### CODE REVIEW

**Comment by retog** [comment]
image
On the left on this PR, on main. Why are no more treatments shown?

**Comment by nleanba** [comment]
> On the left on this PR, on main. Why are no more treatments shown?

This was because the script in `index.html` was not updated and ran into errors. I have fixed it now

File: SynonymGroup.ts [hunk 1]
```diff
@@ -276,7 +99,7 @@ export class SparqlEndpoint {
       } catch (error) {
         if (error instanceof DOMException) {
           // i.e. signal is aborted
-          throw error;
+          return new Promise<SparqlJson>((_, reject) => reject(error));
```
Comment (retog) [inline SynonymGroup.ts]: Why this change, isn't the result exactly the same? If not, could it be simplified to: `return Promise.reject(error);`?

File: SynonymGroup.ts
Comment (retog) [inline SynonymGroup.ts:1]: For consistency within this file, I suggest to use async functions and never the Promise-constructor.

**Review by retog — DISMISSED** [review]
Revert back to throwing exceptions, except there's a good reason.

Otherwise it looks good!

File: SynonymGroup.ts (hunk 1 above)
Comment (nleanba) [reply SynonymGroup.ts]: You are correct, the result is the same. I don't know why I changed this

File: SynonymGroup.ts
Comment (nleanba) [reply SynonymGroup.ts:1]: I have now removed all `new Promise(...)` constructs that resolve instantly, and replaced most of them with `Promise.resolve(...)`.

I deem it unnecessary to convert all function which return promises to async functions, I suggest leaving all instances of `return sparqlEndpoint.getSparqlResultSet(...).then(..., catch);` as-is.

**Pull request merged**